protocol/synthetix/contracts/storage/Pool.sol

```diff
@@ -507,4 +528,25 @@ library Pool {
             );
         }
     }
+
+    function checkPoolCollateralLimit(
+        Data storage self,
+        address collateralType,
+        uint256 collateralAmountD18
+    ) internal view {
+        uint256 maxDeposit = self.collateralConfigurations[collateralType].maxDepositD18;
```

Comment:
`self.vaults[collateralType].currentCollateral()` we can store this in a variable since it's used twice